autoemulate/core/metrics.py

```diff
@@ -0,0 +1,154 @@
+"""Metrics configuration and utilities for model evaluation and tuning."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from functools import partial
+
+import torchmetrics
+
+from autoemulate.core.types import MetricLike
+
+
+class Metric:
+    """Configuration for a single metric.
+
+    Parameters
+    ----------
+    metric : MetricLike
+        The torchmetrics metric class or partial.
+    name : str
+        Display name for the metric.
+    maximize : bool
+        Whether higher values are better. Defaults to True.
+    """
+
+    metric: MetricLike
```

Comment:
As for: https://github.com/alan-turing-institute/autoemulate/pull/883/files#r2444380674
```suggestion
```